common/workunit/workunit.cpp

```diff
@@ -6336,10 +6341,60 @@ IStringVal& CLocalWorkUnit::getSecurityToken(IStringVal &str) const
     return str;
 }
 
-void CLocalWorkUnit::setSecurityToken(const char *value)
+void CLocalWorkUnit::getDistributedAccessToken(IStringVal & datoken) const
 {
     CriticalBlock block(crit);
-    p->setProp("@token", value);
+    datoken.set(p->queryProp("@distributedAccessToken"));
+}
+
+void CLocalWorkUnit::setDistributedAccessToken(const char * datoken)
+{
+    CriticalBlock block(crit);
+//TODO This method probably isn't needed. If so, should I verify it here?
```

Comment:
Haven't looked at enough of who should be able to generate the token... but it should either come from inside workunit dll, or outside, not both.